### DISCUSSION TITLE
Artillery load test runs fine from local machine through command prompt but fails from AWS Code Build

### DISCUSSION THREAD

**Discussion by indrajiit** [artilleryio/artillery#2384] [Artillery]
Hi Team,

My application is deployed in AWS and its a Lambda function integrated with an api endpoint(which i want to test) and I want to do a performance testing on that api endpoint. I have created yaml script file in Artillery and it works fine from my local when i execute and there are no vusers failing but when i create a pipeline in AWS code build and try to start the build it fetches the GIT repo successfully but gives a Timeout error and vusers fail. Can someone help ?

Note:- The application is secured and needs whitelisting of ip's in security groups to run I added my local machine public ip and then it runs fine from local machine . In AWS code build its failing

**Comment by hassy** [comment]
Hi @indrajiit 👋 Impossible to say without more information about the AWS environment. Are CodeBuild jobs allowed to access the application?

**Reply by indrajiit** [reply]
Hi Hassy, You mean to say that if IAM policies are attached to codebuild to access the application ?

**Reply by indrajiit** [reply]
Its able to download the code from the git repo but not executing that I also tried to attach a VPC to the code build and by doing that its not able to download the code as well. Do you think it can be a problem with the private subnet ?

**Reply by hassy** [reply]
@indrajiit it sounds like it's a security/networking issue in your AWS setup. You need to make sure that containers in CodeBuild are able to access the API endpoint. Sorry can't be more specific as it depends on your setup.
